crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -610,6 +630,9 @@ where
                         Some(CurrentAssignment::AugAssign(aug_assign)) => {
                             self.add_definition(symbol, aug_assign);
                         }
+                        Some(CurrentAssignment::For(for_stmt)) => {
+                            self.add_definition(symbol, for_stmt);
```

Comment:
Here's where we'll need to be passing in some node that is specific to the symbol (i.e. the name node), in addition to the overall for-stmt node, more like we do for Assign above (another case where unpacking can happen), so as to distinguish what the Definition is actually for.